ValgrindCI/parse.py

```diff
@@ -130,6 +132,27 @@ def set_source_dir(self, source_dir: Optional[str]) -> None:
         else:
             self._source_dir = None
 
+    def set_substitute_paths(self, substitute_paths: Optional[List[dict]]) -> None:
+        if substitute_paths is not None:
+            self._substitute_paths = substitute_paths
+
+    def set_relative_prefixes(self, relative_prefixes: Optional[str]) -> None:
+        if relative_prefixes is not None:
+            self._relative_prefixes = relative_prefixes
+
+    def substitute_path(self, path: str) -> str:
+        for s in self._substitute_paths:
+            path = path.replace(s.get("from"), s.get("to"))
+        return path
+
+    def relativize(self, path: str) -> str:
+        for p in self._relative_prefixes:
+            if path.startswith(p):
+                path = path.replace(p, "")
+                if path.startswith("/"):
+                    path = path[1:]
```

Comment:
Use òs.path.commonprefix()` and `os.path.relpath()` which are portable across OSes rather than homemade code:
```suggestion
            if os.path.commonprefix([p, path]) == p:
                path = os.path.relpath(path, p)
```